proud-puffins/djangoProject/earlydating/pixel.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+from PIL import Image
+import os
+
+
+def pixelate(filename):
+
+    # Open File
+    absolute = Path.cwd().resolve().parent.joinpath("static", "users_large")
+    img = Image.open(absolute.joinpath(filename))
+
+    # Resize smoothly down to 48x48 pixels
+    imgSmall = img.resize((img.size[0] // 8, img.size[1] // 8), resample=Image.BILINEAR)
+
+    # Scale back up using NEAREST to original size
+    result = imgSmall.resize(img.size, Image.NEAREST)
+
+    # Save
+    result.save(Path.cwd().resolve().parent.joinpath("media", "user_pixel", filename))
+
+
+directory = Path.cwd().resolve().parent.joinpath("static", "users_large")
+
+
+for filename in os.listdir(directory):
+    pixelate(filename)
```

Comment:
I see that this file is not imported anywhere. When you have a Python file in a larger package like this one, individual files are only executed if it is the file that was your entry point, or if it's imported later on. When a file is imported, the whole file is executed from top to bottom one time, and each variable is saved for you to use later. So in this example, you can import the `pixelate` function into any other file and use it as many times as you'd like, and you'll have access to `directory`, but the for loop on lines 25 and 26 will just execute one time whenever you happen to first import this file, and never again after.